common/authorization/oauthAutorizer_test.go

```diff
@@ -181,7 +188,7 @@ func (s *oauthSuite) TestIncorrectPermissionInAttributes() {
 
 func (s *oauthSuite) TestIncorrectDomainInAttributes() {
 	s.att.DomainName = "myotherdomain"
-	authorizer := NewOAuthAuthorizer(s.cfg, s.logger)
+	authorizer := NewOAuthAuthorizer(s.cfg, s.logger, s.domainCache)
```

Comment:
Is domain cache something that changes per user? If that's the case it might be better inside of Attributes (when being called using Authorization). 

I'm implementing Authentication and if I need to use this variable, probably this is the right place then.

Reply:
This is a component that we are going to get some attributes by domainName. 

something like
```
./cadence --do samples-domain domain desc
Name: samples-domain
UUID: cfebaf91-d58d-46fe-9951-345c357332bf
Description:
OwnerEmail:
DomainData: map[ WRITE_GROUPS:group3@indeed.com group4@indeed.com serviceA serviceB READ_GROUPS:group1@indeed.com group2@indeed.com]
```
Get the read or write groups that have the permission, and then compare with the `groups` in the JWT token. 
It's better to do it inside this authroizer because not every implementation needs it. Getting the domain data may have some overhead(very low though, since it's a cache).